source/extensions/io_socket/buffered_io_socket/BUILD

```diff
@@ -0,0 +1,36 @@
+load(
+    "//bazel:envoy_build_system.bzl",
+    "envoy_cc_library",
+    "envoy_extension_package",
+)
+
+licenses(["notice"])  # Apache 2
+
+envoy_extension_package()
+
+envoy_cc_library(
```

Comment:
These libraries should be marked extension libraries.

Also, when there is an actual config to register/load the extension it should be marked alpha so it's not covered by the CVE process until it gets bake time somewhere. What is the plan for how we will register this code?

Reply:
added
```
envoy_cc_extension(
...
    security_posture = "unknown",
    status = "alpha",
```

Reply:
The plan is to extend 
```
DispatcherImpl::createClientConnection(
                                       Network::Address::InstanceConstSharedPtr address,
                                       Network::Address::InstanceConstSharedPtr source_address,
...

so the new io socket can be constructed if the destination address is internal address.

https://github.com/envoyproxy/envoy/pull/13361/files#diff-8a39b55fdec8ce0be24b0126677a41c07207cb5d062d623973990b7be53ac291R154-R155

Reply:
I'm confused. Is this code an extension or is it going to be core code? If an extension how is it going to be configured using typed_config?

/wait

Reply:
We need to build a new extension point that results in creation of these new connection types when specific address types are involved.

Prototype usage can be seen here, but that was before we started going down the extension route:
https://github.com/envoyproxy/envoy/pull/13361/files#diff-70f33a41a52fb3e38be32857512e0de967bf016ee43479ca1530aa86c9843790

Reply:
OK sounds good as long as we are planning on doing it as an extension.

Reply:
Sorry I only answer where do we want to register the where we are going to use this new socket but not how.

I am referring to the fact we have the enough information to locate the BufferedIoSocket.

Follow your question if use typed_config: 
the `Network::Address::Instance` address could translate to proto message address, and then we define new `TypedFactory` for the address proto.

However, I prefer to use `UntypedFactory` which avoid the "Address::Instance" to proto message transform. 
Is UntypedFactory allowed in codebase? I can add a new Factory and test code in this PR